server/public/model/websocket_client.go

```diff
@@ -343,9 +343,10 @@ func (wsc *WebSocketClient) UpdateActiveTeam(teamID string) {
 }
 
 // UpdateActiveThread sets the channel id of the current thread that the user is in.
-func (wsc *WebSocketClient) UpdateActiveThread(channelID string) {
+func (wsc *WebSocketClient) UpdateActiveThread(isThreadView bool, channelID string) {
 	data := map[string]any{
 		"thread_channel_id": channelID,
+		"is_thread_view":    isThreadView,
```

Comment:
Tangential to the changes here but more of a discussion point. Any thoughts in making these payloads properly typed? I know we used this pattern for ages but I wonder whether it would be worth to design something slightly more "stable".

Reply:
The problem is that there is already a high level type `WebsocketRequest`. So any request specific type needs to be embedded in that. Which means you MUST do a type casting atleast once in your workflow. So I don't see a huge benefit here except a bit of type safety on the client side. Since this is all internal to Mattermost, I am thinking it's fine to continue this way.